launcher/modplatform/modrinth/ModrinthPackIndex.cpp

```diff
@@ -55,7 +55,8 @@ void Modrinth::loadIndexedPackVersions(Modrinth::IndexedPack & pack, QJsonArray
         // Find correct file (needed in cases where one version may have multiple files)
         // Will default to the last one if there's no primary (though I think Modrinth requires that
         // at least one file is primary, idk)
-        while (i < files.count()){
+        // NOTE: files.count() is 1-indexed, so we need to subtract 1 to become 0-indexed
+        while (i < files.count() - 1){ 
```

Comment:
I guess this is fine for now. It us kinda ugly though

Reply:
I agree. But it's easier to think about than the previous `files.count() > 1 &&` before that other PR